### PR TITLE
Expose issue stopwatch toggling via API

### LINKED ISSUE #5942: Feature Request: Expose Time Tracker Start and Stop in API
- Gitea version (or commit ref): 1.7.1 (Docker image from Docker Hub)
- Git version: N/A
- Operating system: openSuSE Leap 15.0
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:
N/A

## Description
While it is possible to add time to an issue, it is not possible to trigger the functionality to start or stop logging time on an issue as it is provided by the Web Interface (see screen below) via API (in accordance to _/api/swagger_ in version 1.7.1).

By this feature request I want to suggest to expose this functionality via API as well - including the ability to query if time us currently tracked ("Start" was issued but no "Stop" yet) or not.

Thank you very much for considering this feature request. 

### Concrete Personal Example
Just my personal issue which would be solved by the implementation of this request:

We are managing our tasks and issues via the "issues functionality" in Gitea, which is working very well for us. By doing so, we also use the "Start" and "Stop" time tracking functionality via the Gitea Web Interface (which seems to had been introduced by #967) which comes very handy.

One issue with that is, that when an Gitea issue is closed by mentioning it in the commit comment or pull request (see #462), the time tracking is not stopped. That for itself would not be a problem as a web hook could be registered for issue changes and when called an API call could be issued to stop the time tracking on the issue - if such an API call would be available.

I would suggest not to change the behaviour when  the issue is closed by mentioning it in the commit comment or pull request, as this would allow the maximum flexibility but to allow for automation like outlined above it would be necessary to have the "Start/Stop" functionality exposed via API.

### In General
In general I believe it would make a lot of sense to expose the "Start/Stop" functionality exposed via API as there could be a number of scenarios where this would allow automation for more efficient handling of issues in Gitea.
### Time Tracker in UI for Reference
image

### PR DESCRIPTION
**Pull request by adelowo**
.. #4327 automatically stops the stopwatch but this PR will be useful in case there 
is a need to programmatically toggle the stopwatch.

Fixes https://github.com/go-gitea/gitea/issues/5942

### CODE REVIEW

File: routers/api/v1/repo/issue.go [hunk 1]
```diff
@@ -439,3 +439,65 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
...
+		ctx.Status(403)
+		return
+	}
+
+	if err := models.CreateOrStopIssueStopwatch(ctx.User, issue); err != nil {
+		ctx.ServerError("CreateOrStopIssueStopwatch", err)
+		return
+	}
+
+	ctx.Status(201)
```
Comment (adelowo) [inline routers/api/v1/repo/issue.go:578]: Not sure if an empty response body is really ideal here 🤔🤔

Comment (lafriks) [reply routers/api/v1/repo/issue.go:578]: Empty response is ok

**Review by lafriks — CHANGES_REQUESTED** [review]
I don't actually like toggle method as you are kind of guessing state... I would prefer if API had start & stop methods

**Comment by adelowo** [comment]
Done @lafriks.

File: routers/api/v1/repo/issue.go [hunk 2]
```diff
@@ -439,3 +439,137 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
...
+		return
+	}
+
+	if !ctx.Repo.CanUseTimetracker(issue, ctx.User) {
+		ctx.Status(403)
+		return
+	}
+
+	if models.StopwatchExists(ctx.User.ID, issue.ID) {
+		ctx.Error(403, "StopwatchExists", "a stopwatch has already been started for this issue")
```
Comment (lafriks) [inline routers/api/v1/repo/issue.go]: 409 Conflict would be better status code for this

File: routers/api/v1/repo/issue.go [hunk 3]
```diff
@@ -439,3 +439,137 @@ func UpdateIssueDeadline(ctx *context.APIContext, form api.EditDeadlineOption) {
...
+	}
+
+	if !ctx.Repo.CanUseTimetracker(issue, ctx.User) {
+		ctx.Status(403)
+		return
+	}
+
+	if !models.StopwatchExists(ctx.User.ID, issue.ID) {
+		ctx.Error(403, "StopwatchExists", "cannot stop a non existent stopwatch")
+		return
```
Comment (lafriks) [inline routers/api/v1/repo/issue.go:570]: Same here

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by lafriks — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Pull request merged**